app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/LocalLibraryFragment.kt

```diff
@@ -393,4 +413,46 @@ class LocalLibraryFragment : BaseFragment() {
   private fun navigateToLocalFileTransferFragment() {
     requireActivity().navigate(R.id.localFileTransferFragment)
   }
+
+  private fun shouldShowRationalePermission() =
+    ActivityCompat.shouldShowRequestPermissionRationale(
+      requireActivity(),
+      Manifest.permission.READ_EXTERNAL_STORAGE
+    )
+
+  private fun isPermissionDenied(grantResults: IntArray) =
+    grantResults[0] == PackageManager.PERMISSION_DENIED
+
+  private fun readStorageHasBeenPermanentlyDenied(grantResults: IntArray) =
+    isPermissionDenied(grantResults) &&
+      !shouldShowRationalePermission()
+
+  override fun onRequestPermissionsResult(
```

Comment:
Have you considered modernizing using the simpler https://developer.android.com/reference/androidx/activity/result/contract/ActivityResultContracts.RequestPermission ?